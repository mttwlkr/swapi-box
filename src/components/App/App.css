body {
  background-color:#141414;
}


.header-background {
  background-image: url('https://i.pinimg.com/originals/a3/78/50/a3785076c32f59bb2984407c415a231a.jpg');
  background-repeat: no-repeat;
  background-position: 111% center;
/*  height: 300px;*/
}


/*.header-background {
  background-image: url('https://images.unsplash.com/photo-1504966981333-1ac8809be1ca?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=9a1325446cbf9b56f6ee549623a50696&auto=format&fit=crop&w=2850&q=80');
  background-size: cover;
  background-position-y: 42%;
  background-repeat: no-repeat;
  height: 300px;
}
*/

