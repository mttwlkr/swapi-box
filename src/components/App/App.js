import React, { Component } from 'react';
import Nav from '../Nav/Nav.js';
import Main from '../Main/Main.js';
import Header from '../Header/Header.js'
// import fetchData from '../../cleaners/fetchData.js'
import './App.css';

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      scroll: '',
      films: '',
      people: '',
      vehicles: '',
      planets: '',
      error: false
    }
    this.getMovie = this.getMovie.bind(this)
  }

  getMovie() {
    fetch('https://swapi.co/api/films')
      .then(response => response.json())
      .then(data => {
        if(data.results) {
          this.setState({ scroll: data.results[Math.floor((Math.random() * 7) + 1)].opening_crawl })
        } else {
          throw Error("Error 404")
        }
      })
    .catch(error => this.setState({error: true}))
  }

  componentDidMount() {
    this.getMovie()
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Nav />
        <Main scroll={this.state.scroll}/>
      </div>
    );
  }
}

export default App;

