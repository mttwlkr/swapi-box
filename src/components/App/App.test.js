import React from 'react';
import { shallow } from 'enzyme';
import App from './App.js';
import { getAPI,
  parseFunc,
  cleanVehicles } from '../../cleaners/cleaner.js';

import {
  mockAPIVehicleData, 
  mockCleanCardData } from '../../cleaners/mocks.js';


describe('App', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />, {disableLifecycleMethods: true});
  });

  it('should render without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update cards state if favorite cards length', () => {
    expect(wrapper.state('cards').length).toEqual(0);
    wrapper.setState({favorites: [mockCleanCardData]});
    wrapper.instance().showFavorites();
    expect(wrapper.state('cards').length).toEqual(1);
  });

  it('should add a favorited card to favorites state array', () => {
    expect(wrapper.state('favorites').length).toEqual(0);
    wrapper.instance().handleFavorite(mockCleanCardData);
    expect(wrapper.state('favorites').length).toEqual(1);
  });

  it('should not add duplicate card to favorites state', () => {
    expect(wrapper.state('favorites').length).toEqual(0);
    wrapper.setState({favorites: [mockCleanCardData]});
    wrapper.instance().handleFavorite(mockCleanCardData);
    expect(wrapper.state('favorites').length).toEqual(0);
  });

  it('should change the isActive prop of card to true if not in state', () => {
    expect(mockCleanCardData.isActive).toEqual(false);
    wrapper.instance().handleFavorite(mockCleanCardData);
    expect(mockCleanCardData.isActive).toEqual(true);
  })

  it('should send an alert if favorites array is empty', () => {
    wrapper.setState({favorites: []});
    window.alert = jest.fn();
    wrapper.instance().showFavorites()
    expect(window.alert).toHaveBeenCalled()
  })

  it('should set card state to favorites if favorites array has length', () => {
    wrapper.setState({favorites: [mockCleanCardData]});
    wrapper.instance().showFavorites()
    expect(wrapper.state('cards').length).toEqual(1)
  })  

  it('should change the isActive prop of card to false if its in state', () => {
    wrapper.setState({favorites: [mockCleanCardData]});
    expect(mockCleanCardData.isActive).toEqual(true);
    wrapper.instance().handleFavorite(mockCleanCardData);
    expect(mockCleanCardData.isActive).toEqual(false);
  })

  it.skip('should invoke the handleFetch method', async () => {
    const mockEvent = {target: {name: 'vehicles'}};
    const handleFetch = jest.spyOn(wrapper.instance(), 'handleFetch');
    wrapper.instance().handleClick(mockEvent);
    expect(handleFetch).toHaveBeenCalled();
  });

  it.skip('should set cards state when handleFetch is invoked', async () => {
    expect(wrapper.state('cards').length).toEqual(0);
    wrapper.instance().handleFetch('vehicles');
    window.fetch = jest.fn().mockImplementation( () => ({
      status: 200,
      json: () => new Promise((resolve) => {
        resolve({
          mockAPIVehicleData
        });
      })
    }));
    await cleanVehicles(mockAPIVehicleData);
    expect(wrapper.state('cards').length).toEqual(10);
  });
});
