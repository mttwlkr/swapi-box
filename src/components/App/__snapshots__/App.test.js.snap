// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-background"
>
          <Header
                    controlFunction={[Function]}
                    value={Array []}
          />
          <Nav
                    controlFunction={[Function]}
          />
</div>,
        <Main
          cards={Array []}
          controlFunction={[Function]}
          scroll={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              controlFunction={[Function]}
              value={Array []}
/>,
            <Nav
              controlFunction={[Function]}
/>,
          ],
          "className": "header-background",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "controlFunction": [Function],
              "value": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "controlFunction": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [],
          "controlFunction": [Function],
          "scroll": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-background"
>
            <Header
                        controlFunction={[Function]}
                        value={Array []}
            />
            <Nav
                        controlFunction={[Function]}
            />
</div>,
          <Main
            cards={Array []}
            controlFunction={[Function]}
            scroll={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                controlFunction={[Function]}
                value={Array []}
/>,
              <Nav
                controlFunction={[Function]}
/>,
            ],
            "className": "header-background",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "controlFunction": [Function],
                "value": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "controlFunction": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": Array [],
            "controlFunction": [Function],
            "scroll": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
